odl/operator/default_ops.py

```diff
@@ -1325,13 +1355,129 @@ def __init__(self, space):
         uniform_discr(0.0, 1.0, 2).element([ 5.,  2.])
         """
         real_space = space.real_space
-        linear = (space == real_space)
-        super(ComplexModulus, self).__init__(space, real_space, linear=linear)
+        super(ComplexModulus, self).__init__(space, real_space, linear=False)
 
     def _call(self, x):
         """Return ``self(x)``."""
         return (x.real ** 2 + x.imag ** 2).ufuncs.sqrt()
 
+    def derivative(self, x):
+        r"""Return the derivative operator.
+
+        Note that the returned operator (``self``) is the derivative in the
+        Gâteaux sense, but not in the Fréchet sense since it is not linear.
+        Nevertheless, it has an `adjoint`.
+
+        Parameters
+        ----------
+        x : `domain` element
+            Point in which to take the derivative.
+
+        Examples
+        --------
+        Derivative operator:
+
+        >>> c2 = odl.cn(2)
+        >>> op = odl.ComplexModulus(c2)
+        >>> op([3 + 4j, 2])
+        rn(2).element([ 5.,  2.])
+        >>> deriv = op.derivative([3 + 4j, 2])
+        >>> deriv.domain
+        cn(2)
+        >>> deriv.range
+        rn(2)
+        >>> deriv([2 + 3j, 4j])  # [5 / 5, 4 / 2]
```

Comment:
Is this really correct?

If I write z_1 = x_1 + i y_1, z_2 = x_2 + i y_2, then
f(z_1, z_2) = f(x_1, y_1, x_2, y_2) = [|z_1| \\ |z_2|], the Jacobian is (when regarded as a mapping ℝ^4 → ℝ²)
df(z_1, z_2) = [x_1/|z_1|, y_1/|z_1|, 0, 0 \\ 0, 0, x_2/|z_2|, y_2/|z_2|]
and for x_1 = 3, y_1 = 4, x_2 = 2, y_2 = 0, when multiplied with [2 \\ 3 \\ 0 \\ 4], I get [18/5 \\ 0]. What is my mistake?

Reply:
Oh crap, I have forgotten a factor from the inner derivative. :hankey: